src/application/service/users/GetExistingOrCreateNewUser.test.ts

```diff
@@ -3,10 +3,10 @@ import database from '../../../database';
 import { GetExistingOrCreateNewUser } from './GetExistingOrCreateNewUser';
 import { cleanupDatabase } from '../../../test/cleanupDatabase';
 import { anEmail, aNewUser } from '../../../test/domainFactories';
-import { USER } from '../../../domain/model/authentication/Roles';
+import { ADMIN, USER } from '../../../domain/model/authentication/Roles';
 
 describe('GetExistingOrCreateNewUser', () => {
-  const getExistingOrCreateNewUser = new GetExistingOrCreateNewUser(userRepository, roleRepository);
+  const getExistingOrCreateNewUser = new GetExistingOrCreateNewUser(userRepository, roleRepository, true);
```

Comment:
Where is the logic that turns setup mode off after the first user is created?

Reply:
It is configurable via an environment variable.

Reply:
Any reason we don't want to do it with a more enforced way? e.g, when we create the user we check if there are any users. If not, we create the superuser. Seems a bit more foolproof

Reply:
What do you imagine would go wrong? Forgetting to disable it? 

Initially I also wanted to do it with the query but: 
1. I couldn't justify running an additional query on every user creation for all the users just for something that we need for a few moments in the beginning. 
2. This more is more flexible. It allows you to have more admins in the beginning without much effort with API calls. 
3. By default the mode is disabled so you would have to want to enable it.